src/merkle/smt/full/mod.rs

```diff
@@ -130,6 +130,11 @@ impl Smt {
         <Self as SparseMerkleTree<SMT_DEPTH>>::open(self, key)
     }
 
+    /// Returns a boolean value indicating whether the SMT is empty.
+    pub fn is_empty(&self) -> bool {
+        self.leaves.is_empty()
```

Comment:
I wonder if the check should be something like `self.root == Self::EMPTY_ROOT`?

Reply:
But will it be incorrect to check the emptiness of the thee by the number of the leaves?

Reply:
They should match. I would maybe even put a `debug_assert!()` there. But the tree is for sure empty if the root of the tree is the same as `Self::EMPTY_ROOT`.